### PR TITLE
convert network accessors to functions

### PR DESCRIPTION
**Pull request by walmat**
## What changed (plus any additional context for devs)
Okay so runtime support wasn't working because our accessors for backendNetworks data were all objects that only read from queryData once and never again on successive calls. I've converted all the accessors to be functions, so now we read up-to-date information when accessing. 

Intro: `backendNetworksStore`.
- this store is a barebones Zustand store, and acts as an access point for the up-to-date backendNetworks data. 
- it stores a shared value version and a non-shared value version

This PR also introduces a couple worklet functions to make swaps behave nicely. The only caveat is for the worklet functions, **you must pass the backend network data in as a parameter**. This is necessary because we cannot call `getState` inside of a worklet.

## Screen recordings / screenshots
N/A as it's just functional changes

## What to test
Test all critical paths

**would be nice to be able to coordinate some backend change to see them reflected in real-time**

### CODE REVIEW

File: src/chains/index.ts [hunk 1]
```diff
@@ -8,62 +8,93 @@ import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
...
-
-export const chainsIdByName: Record<string, number> = backendNetworks.networks.reduce(
-  (acc, backendNetwork: BackendNetwork) => {
-    acc[backendNetwork.name] = parseInt(backendNetwork.id, 10);
-    return acc;
-  },
-  {} as Record<string, number>
-);
+const getBackendNetworks = (): BackendNetworksResponse => {
+  return queryClient.getQueryData<BackendNetworksResponse>(backendNetworksQueryKey()) ?? buildTimeNetworks;
```
Comment (walmat) [inline src/chains/index.ts:12]: This won't work for parts of the app that require UI thread execution, specifically https://github.com/rainbow-me/rainbow/blob/5d5dfed90bbef1b9b39534c7528f09f095997a3d/src/__swaps__/screens/Swap/components/TokenList/ChainSelection.tsx#L51

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/215ea07d0c491555881cd94fd7f987854088faca.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/215ea07d0c491555881cd94fd7f987854088faca.ipa) for 215ea07d0c491555881cd94fd7f987854088faca

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     "install-pods-fast": "cd ios && bundle exec pod install && cd ..",
     "install-pods-fast-jsc": "cd ios && bundle exec env USE_HERMES=NO pod install && cd ..",
     "install-pods-no-flipper": "cd ios && bundle exec env SKIP_FLIPPER=true pod install --repo-update && cd ..",
-    "ios": "react-native run-ios --simulator='iPhone 16 Pro'",
+    "ios": "react-native run-ios --simulator='iPhone 15 Pro'",
```
Comment (walmat) [inline package.json:44]:
```suggestion
    "ios": "react-native run-ios --simulator='iPhone 16 Pro'",
```

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/f751ea82a1950491bcbd958f06467ceb2bd494be.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/f751ea82a1950491bcbd958f06467ceb2bd494be.ipa) for f751ea82a1950491bcbd958f06467ceb2bd494be

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's, QA Passed 👍🏽

**Comment by derHowie** [comment]
Yo, @walmat. Looks like the linter in complaining about `SpeedUpAndCancelSheet`.

**Comment by walmat** [comment]
> Yo, @walmat. Looks like the linter in complaining about `SpeedUpAndCancelSheet`.

the linter is my best friend. Good catch! Taking a look now

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/af717544494ef15163f47f55c1c362e7f41bb619.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/af717544494ef15163f47f55c1c362e7f41bb619.ipa) for af717544494ef15163f47f55c1c362e7f41bb619

**Review by ibrahimtaveras00 — APPROVED** [review]
spot checks were good 👍🏽

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/17447c2b41e9cc6bb45236495def05de99c19929.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/17447c2b41e9cc6bb45236495def05de99c19929.ipa) for 17447c2b41e9cc6bb45236495def05de99c19929

**Comment by derHowie** [comment]
@walmat can you check out the conflicts here?

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/65455b96ba39c1d4e05d8bc7d91df40df0dd5eb2.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/fix-runtime-networks/65455b96ba39c1d4e05d8bc7d91df40df0dd5eb2.ipa) for 65455b96ba39c1d4e05d8bc7d91df40df0dd5eb2

**Pull request merged**